### PR TITLE
Parser improvements

### PR DESCRIPTION
**Pull request by InnocentZero**
- Fixes match parsing to include optional comma at the end of match arm body.
- Parses unit (`()`) properly now
- Fixed parsing, and added resolution and typechecking for alias.

Depends on #19 so that needs to get merged first.

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -1314,11 +1333,24 @@ impl Parser {
         let span_start = self.position();
         self.keyword(b"while");
 
+        if self.is_operator() {
+            // TODO: flag parsing
+            self.next();
```
Comment (kubouch) [inline src/parser.rs:1338]: Maybe throw an error here with a WIP note instead of silently ignoring flags.

**Comment by kubouch** [comment]
Haven't checked it too thoroughly, but also seems good. Flags with aliases we can add once we have proper signatures etc.

**Comment by InnocentZero** [comment]
I was going to add flags next, but it seems to be involved deeply with the parser and since you're rewriting the lexer which will move parts of the parser, I'll wait for it to settle a little bit. For now, I'll rebase once #19 is accepted and add the error for flags.

**Comment by InnocentZero** [comment]
I'm closing this in favour of #29 

This one became a mess after my previous PR was merged and I wasn't able to resolve it easily.

**Pull request closed without merging**